ui/src/tempVars/components/MeasurementsTemplateBuilder.tsx

```diff
@@ -0,0 +1,162 @@
+import React, {PureComponent} from 'react'
+import _ from 'lodash'
+
+import {
+  TemplateBuilderProps,
+  RemoteDataState,
+  TemplateValueType,
+} from 'src/types'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import Dropdown from 'src/shared/components/Dropdown'
+import {showDatabases, showMeasurements} from 'src/shared/apis/metaQuery'
+import parseShowDatabases from 'src/shared/parsing/showDatabases'
+import parseShowMeasurements from 'src/shared/parsing/showMeasurements'
+import TemplateMetaQueryPreview from 'src/tempVars/components/TemplateMetaQueryPreview'
+import DropdownPlaceholder from 'src/shared/components/DropdownPlaceholder'
+
+interface State {
+  databases: string[]
+  databasesStatus: RemoteDataState
+  selectedDatabase: string
+  measurements: string[]
+  measurementsStatus: RemoteDataState
+}
+
+@ErrorHandling
+export default class MeasurementsTemplateBuilder extends PureComponent<
+  TemplateBuilderProps,
+  State
+> {
+  constructor(props) {
+    super(props)
+
+    const selectedDatabase = _.get(props, 'template.query.db', '')
+
+    this.state = {
+      databases: [],
+      databasesStatus: RemoteDataState.Loading,
+      selectedDatabase,
+      measurements: [],
+      measurementsStatus: RemoteDataState.Loading,
+    }
+  }
+
+  public async componentDidMount() {
+    await this.loadDatabases()
+    await this.loadMeasurements()
+  }
+
+  public render() {
+    const {
+      databases,
+      databasesStatus,
+      selectedDatabase,
+      measurements,
+      measurementsStatus,
+    } = this.state
+
+    return (
+      <div className="temp-builder measurements-temp-builder">
+        <div className="form-group">
+          <label>Meta Query</label>
+          <div className="temp-builder--mq-controls">
+            <div className="temp-builder--mq-text">SHOW MEASUREMENTS ON</div>
+            <DropdownPlaceholder rds={databasesStatus}>
+              <Dropdown
+                items={databases.map(text => ({text}))}
+                onChoose={this.handleChooseDatabaseDropdown}
+                selected={selectedDatabase}
+                buttonSize=""
+              />
+            </DropdownPlaceholder>
+          </div>
+        </div>
+        <TemplateMetaQueryPreview
+          items={measurements}
+          loadingState={measurementsStatus}
```

Comment:
now we're seeing a conflict between `State` and `Status` :) what do you think about `loadingStatus`?